static_frame/core/frame.py

```diff
@@ -832,6 +832,15 @@ def __init__(self,
             # will have shape of 0,0
             self._blocks = TypeBlocks.from_none()
 
+        elif data is None and (
+                hasattr(index, '__len__') or hasattr(columns, '__len__')):
```

Comment:
I think that all of these `hasattr(XXX, '__len__')` calls on lines 836-838 should instead be `XXX is not None`, and `index` and `columns` should be converted to tuples before line 837.

Not only does it read better, but this PR currently breaks support for generators as indices or columns! These constructions work now:

```py
>>> sf.Frame(None, index=(i for i in range(3)), columns=("A",))
<Frame>
<Index>  A        <<U1>
<Index>
0        None
1        None
2        None
<object> <object>
```

```py
>>> sf.Frame(None, index=("A",), columns=(i for i in range(3)))
<Frame>
<Index> 0        1        2        <object>
<Index>
A       None     None     None
<<U1>   <object> <object> <object>
```

But with this patch applied, the resulting `Frame` is mangled:

```py
>>> sf.Frame(None, index=(i for i in range(3)), columns=("A",))
<Frame>
<Index>  A        <<U1>
<Index>
0        <object>
1
2
<object>
```

```py
>>> sf.Frame(None, index=("A",), columns=(i for i in range(3)))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/brandtbucher/Desktop/GitHub/temp/static-frame/static_frame/core/frame.py", line 906, in __init__
    len(self._columns), col_count))
RuntimeError: Columns has incorrect size (got 3, expected 0)
```